harvester_e2e_tests/integration/test_volumes.py

```diff
@@ -72,3 +74,68 @@ def _wait_for_volume_ready():
         timeout=request.config.getoption('--wait-timeout'))
     assert success, 'Timed out while waiting for volume to be ready.'
     return pvc_json
+
+
+@pytest.mark.volumes
+@pytest.mark.p1
+def test_create_volume_backing_image(api_client, unique_name, opensuse_image,
+                                     wait_timeout, request):
+    """
+    1. Create a new image from URL
+    2. Check that it is created succesffully.
+    3. Create a new volume with the image_id of the previous image
+    4. Check that the new image is created
+    5. Delete image and volume
+    """
+
+    code, image_data = api_client.images.create_by_url(unique_name, opensuse_image.url)
+    if code != 201:
+        raise AssertionError(
+            f"Failed to create image {unique_name} from URL got\n"
+            f"Creation got {code} with {image_data}"
+        )
+
+    assert 201 == code, (code, image_data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.images.get(unique_name)
+        image_conds = data.get('status', {}).get('conditions', [])
+        if len(image_conds) > 0:
+            break
+        sleep(3)
+    # This waits for the import to finish
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.images.get(unique_name)
+        image_conds = data.get('status', {}).get('conditions', [])
+        if image_conds[0]['reason'] == 'Imported':
```

Comment:
```suggestion
        if len(image_conds) != 0 and image_conds[0]['reason'] == 'Imported':
```
There could be a chance that if we're waiting in the while loop, since we set `image_conds` to ultimately be an empty list, that by the type the evaluation of:
```python
if image_conds[0]['reason'] == 'Imported':
```
It could fire back an error of:
```python
IndexError: list index out of range
```
Which would bubble up and fail the test since we're not currently catching the Exception.
When it might just mean that the test wouldn't fail, because we were just waiting for the conditions - and it just happens in the first iteration we didn't get any, so we set as an empty list - but maybe 3-4 iterations down the line we would get 'conditions' and we wouldn't end up setting it as an empty list.

I know earlier in the test we have that logic block of:
```python

   while endtime > datetime.now():
        code, data = api_client.images.get(unique_name)
        image_conds = data.get('status', {}).get('conditions', [])
        if len(image_conds) > 0:
            break
        sleep(3)
```
So I guess it could also be written something like:
```suggestion
        if len(image_conds) > 0 and image_conds[0]['reason'] == 'Imported':
```
too :smile:

Reply:
Excellent point. I'll make the change. That's more concise and  less prone to timeouts.

Reply:
Changed.